packages/kbn-test/src/functional_test_runner/functional_test_runner.ts

```diff
@@ -63,7 +63,7 @@ export class FunctionalTestRunner {
 
       // validate es version
       if (providers.hasService('es')) {
-        const es = (await providers.getService('es')) as unknown as EsClient;
+        const es = (await providers.getService('esWithoutSystemIndices')) as unknown as EsClient;
```

Comment:
Can you help me understand the intent here? We're not currently ensuring that the `esWithoutSystemIndices` service is loaded into the FTR, but this seems to work... Maybe we should create our own client?

Reply:
If we're the only consumer of `esWithoutSystemIndices` we should just create the client here instead of using a service at all.

Reply:
@spalger The `esWithoutSystemIndices` is only used for two tasks: do the initial ES version checking and create the `system_indices_superuser`, which is then used by the `es` service for the rest of the tests. I've implemented it like this as it was the closest thing to the existing code that I could come up with, but happy to adjust. If we create the client here directly, would we do it again for the user creation?

Reply:
Submitted https://github.com/pheyos/kibana/pull/1 with suggested changes which creates independent clients in both places, but does it using a new helper which applies the copy-pasted logic implemented in several places as necessary.

Reply:
Thanks @spalger! I think these are good changes and merged your commits.

Reply:
Circling back to my first comment in this PR: do you know if we still need the user/role creation in the native realm anymore (https://github.com/elastic/kibana/blob/main/packages/kbn-es/src/utils/native_realm.js#L132)? AFAICTwe could remove it from there, but not sure if there's more happening under the hood that I'm not aware of.

Reply:
I think we could move this native realm topic to a separate investigation / potential follow-up PR to not delay this fix and unblock cloud testing.